core/src/main/java/org/elasticsearch/index/mapper/core/KeywordFieldMapper.java

```diff
@@ -128,6 +120,8 @@ public KeywordFieldMapper build(BuilderContext context) {
                 } else if (propName.equals("ignore_above")) {
                     builder.ignoreAbove(XContentMapValues.nodeIntegerValue(propNode, -1));
                     iterator.remove();
+                } else if (TypeParsers.parseNorms(builder, propName, propNode, parserContext)) {
```

Comment:
I had done this with #16991 in mind so that strings can be automatically upgraded. I have also thought about upgrading the norms settings in StringFieldMapper.Parser so that text and keyword don't have to know about old norms settings, but this could make errors hard to understand since they would complain about the upgraded settings instead of the original ones that have been provided by the user.